### ISSUE TITLE
RFC: zstandard compression

### ISSUE THREAD

**Issue by kdmurray91** [seqan/seqan#1922]
HI all,

[zstd](https://github.com/facebook/zstd) is a relatively new compression algorithm with incredible speed and compression ratios. I've been using zstd (via stream redirection) on fastq files extensively, with great effect (and speed ups compared to gzip).

Would zstd compression be considered for inclusion within seqan?

Cheers,
K

**Comment by h-2** [comment]
I would like to support more compression formats, as well. My priority was lz4, because its so fast, its basically transparent. You could even use it for im-memory compression of strings…
On the other end of the speed VS compression-ratio scale there is XZ which would also be nice to have. TBH my analysis of the compression landscape was:
 * if you want speed, take lz4
 * if you want high ratio, take xz (lzma)
 * if you want compatibility, take gz

Even from https://github.com/facebook/zstd the advantages of zstd over lz4 don't seem immediately clear to me in this picture, but maybe I have to do more reading.

In any case, adding a compression format to SeqAn2 is quite an undertaking, but for SeqAn3 we will likely switch to Boost::IOStreams. Boost::IOStreams have in-house support for gz and bz2, and there are third-party adapters for lz4 and lzma.

**Comment by kdmurray91** [comment]
@h-2 That makes perfect sense. The rationale for my use of zstd is, a) it seems to hit the sweet spot between compression ratio and decompression speed with the hardware I use most often, and b), you can tweak compression to be as high (and slow) as xz while having a massively increased decompression speed, for situations where disc space is limited. You're right that between lz4, xz and gz, most of these bases are largely covered (save the faster decompression speed, which for my use cases is actually more important).

**Comment by h-2** [comment]
True, it seems to preserve the good decompression speed even at higher compression ratios. We will keep this issue open for seqan 3.
